### DISCUSSION TITLE
Issues with OpenRGB-0.8

### DISCUSSION THREAD

**Discussion by enku** [jath03/openrgb-python#64] [Q&A] [answered]
[Moved over from Issues]

I have a setup where I have a bunch of "devices" (not sure if that's the right OpenRGB term) all plugged into a single hub which is then plugged into an RGB port on my MSI motherboard.  These devices have multiple "LEDs" on them and I really just want "all LEDs on all devices the same color".  On OpenRGB-0.7 the following does what I want:

```python
for device in openrgb.ee_devices:
    rgb_color = RGBColor(randint(0, 255), randint(0, 255), randint(0, 255))
    device.set_mode("direct")
    device.set_color(rgb_color)
```

When I upgrade to OpenRGB-0.8 however it seems that only 2 LEDs on each device gets set to the color and the other LEDs are off.  Oddly it seems to be 2, not 1 which confuses me but perhaps they're somehow "mirrored".  I've tried `set_colors([rgb_color] * len(device.leds)` but the result is the same.

From what I can tell (not much since I'm fairly new to the ARGB game), one of the following is occurring, ordered by likeliness.

- The above code is incorrect for what I want but just so happens to work with OpenRGB-0.7
- Something changed in OpenRGB-0.8 and openrgb-python does not yet work with this change
- There is a regression bug in OpenRGB-0.8
- There is something wrong with the hardware

As I said likely there is something I'm doing wrong in the code, but if that's the case I'm not sure what it is (or why it worked before).

openrgb-python-0.2.15 on Linux.

**Comment by jath03** [comment, marked as answer]
Maybe somehow the size of the ARGB zone got changed with the update?  Try resizing the zone in openrgb.

If you turn on the LED view in openrgb and then run the python code, do the colors change in the led view as expected?
If you add a small sleep after each API call, does anything change?

**Reply by enku** [reply]
I'm not familiar with what zones are. Let me look that up and give it a try.

**Reply by enku** [reply]
This seems to work for me:

```python
for device in openrgb.ee_devices:
    device.set_mode("direct")
    led_count = len(device.leds)

    for zone in device.zones:
        zone.resize(led_count)

    rgb_color = RGBColor(randint(0, 255), randint(0, 255), randint(0, 255))
   
    device.set_color(rgb_color)
```
